### ISSUE TITLE
Currency numeric type?

### ISSUE THREAD

**Issue by GatoRat** [miloyip/rapidjson#345]
I'm working on a C++ application which exchanges data with a Java app using JSON. On the Java side, they used a currency class. The data is written as floating point value in the JSON. We handle this by reading it as a double and then converting it to pennies. And visa versa when going the other way. This works, but the use of double for currency annoys me.

I've been debating possible solutions. The most flexible would be to add the ability to "ReadAsString" and WriteAsString, which would treat a numeric value as a string with the restriction that the value must ultimately be numeric.

Perhaps there is already a way to do this, in which case, educate me.

**Comment by miloyip** [comment]
Currently RapidJSON does not support this feature. The feature you request may be like "validating JSON numbers but not converting them to `double`s", and vice versa.

I am not sure whether this feature is a common need. If so, we can consider adding this in the coming versions. My idea for that is adding a parse flag, related SAX events and DOM APIs.

A simple solution is to serialize the currency as JSON string type instead of number type. And then you do the format validation and conversion in the application side.

I am open for discussion.

**Comment by GatoRat** [comment]
I agree that the currency values should have been sent as strings. Alas, the team we are interacting with can't change due to their existing clients. (Another solution, which we were doing in our own code, is to store currency as 64-bit integers and apply a denominator.)

I agree that this may not be a common need. However, I'll still ponder various solutions.

**Comment by miloyip** [comment]
For writing, it may be possible to use `Writer::RawValue()` in #543 . But there is no counterpart in `Reader/Document` now.
